### ISSUE TITLE
Wrong list of labels in an introduction notebook

### ISSUE THREAD

**Issue by VHursevich** [tensorflow/tensorflow#88643]
### Issue type

Bug

### Have you reproduced the bug with TensorFlow Nightly?

Yes

### Source

source

### TensorFlow version

tf 2.18

### Custom code

Yes

### Python version

3.11.11

### Current behavior?

[https://github.com/tensorflow/docs/blob/master/site/en/tutorials/audio/simple_audio.ipynb](url)

x_labels = ['no', 'yes', 'down', 'go', 'left', 'up', 'right', 'stop']

![Image](https://github.com/user-attachments/assets/9baa362a-366e-460d-aa4a-9c8b20a22bb4)

Correct list of labels depends on system.

For Ubuntu 22.04 correct labels: 
x_labels = ['down', 'go', 'left', 'no', 'right', 'stop', 'up', 'yes'] 

![Image](https://github.com/user-attachments/assets/8193147a-ac05-46d7-87a3-d132f5e51db1)

### Standalone code to reproduce the issue

```shell
[https://github.com/tensorflow/docs/blob/master/site/en/tutorials/audio/simple_audio.ipynb](url)
```

### Relevant log output

```shell

```

**Comment by meirtauber** [comment]
Is the problem that the label order changes for different operatiing systems? Or that you want the order of the labels to change with the operating system. If its the first option, you could explicitly by adding 
`sorted(os.listdir(data_dir))`

**Comment by kamaleshwar-8** [comment]
The issue stems from how labels are handled in the original code. I've fixed this issue by ensuring consistent label ordering throughout the pipeline:

Python : 



      def main():
          # Get commands (classes)
          commands = []
          for item in data_dir.iterdir():
              if item.is_dir() and item.name not in ['README.md', '.DS_Store']:
                  commands.append(item.name)
          commands = sorted(commands) # Sort for consistency
  
      # Load and preprocess dataset
      train_ds, val_ds = tf.keras.utils.audio_dataset_from_directory(
          directory=data_dir,
          batch_size=64,
          validation_split=0.2,
          seed=0,
          output_sequence_length=16000,
          subset='both'
      )
      
      # Get label names
      label_names = np.array(train_ds.class_names)
      
      # Save label names properly
      labels_list = label_names.tolist() if hasattr(label_names, 'tolist') else label_names
      with open("label_names.json", "w") as f:
          json.dump(labels_list, f)


And for prediction, I've implemented proper preprocessing and consistent label handling:


# Proper prediction code

    def preprocess_audiobuffer(waveform):
        """Preprocess audio buffer for model prediction"""
        # Normalize from [-32768, 32767] to [-1, 1]
        waveform = waveform / 32768
        waveform = tf.convert_to_tensor(waveform, dtype=tf.float32)
        spectrogram = get_spectrogram(waveform)
        # Add batch dimension
        spectrogram = tf.expand_dims(spectrogram, 0)
        return spectrogram
  
# Test prediction with consistent label handling

    prediction = trained_model(spec, training=False)
    probabilities = tf.nn.softmax(prediction[0])
    predicted_index = tf.argmax(probabilities).numpy()
    predicted_word = label_names[predicted_index]
